### PR TITLE
Add stricter validation for api key expiration time

### PR DESCRIPTION
**Pull request by jfreden**
This is a follow up from: https://github.com/elastic/elasticsearch/pull/103453

> Your PR prompted me to think about this again: it's kinda bogus that in the create API you can set 0 or -1 for expiration -- we've talked in the past about preventing this but didn't get around to it. I like being consistent between APIs, but I don't think we want to inherit API flaws. WDYT about preventing non-positive values for expiration for the update APIs?

### CODE REVIEW

**Review by n1v0lg — COMMENTED** [review]
Wow sorry I really let this one sit way too long. I have a suggestion around moving the validation logic to the request classes instead.

@jakelandis I wanted to get a sanity check on _not_ treating this as a breaking change. I don't think it is, but it's good to call out and discuss explicitly:

Today, it's possible to specify a 0 or negative expiration time when creating an API key. This means the API key is invalid immediately. That really doesn't make any sense -- the only reason I see this happening is users misunderstanding `expiration` (e.g., maybe they assume `-1` means no expiration?). 

This PR adds validation that rejects bogus (0 or negative) expiration times. For the update API key APIs this is non-controversial since we're adding the validation in the same minor as we added the field. 

We're also adding this check to the create API key API; there the field has been around for a long time. Like I said, I don't see a valid use case for supplying bogus values, and adding the validation is a small but nice improvement. So I'd say, let's move ahead with it and not treat is as a breaking change. Would you agree here?

Thanks!

**Comment by jakelandis** [comment]
> I wanted to get a sanity check on not treating this as a breaking change.

This is a breaking change. It might be one that goes primarily unnoticed, but it is still breaking. Introduction of this change would be subject to [REST API compatibility](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-api-compatibility.html). We could go through the process of breaking change (approval, deprecation, compatibility) but not sure if this one is worth the effort since given that due to our history, the bar is pretty high.

**Comment by jfreden** [comment]
Thanks @jakelandis and @n1v0lg ! So our options are: 
1. Skip this validation entirely.
2. Only do this validation in the update api and lose consistency in the validation between the update and create APIs. 
3. Go through the breaking change process.  

I think I'm leaning towards (1). What do you think @jakelandis and @n1v0lg ?

**Comment by n1v0lg** [comment]
@jakelandis noted, thanks! My rationale for proposing this change is that it feels more like a bug fix -- essentially, the only thing you get out of the "BWC"-behavior is unusable API keys. However, this is subjective and strictly speaking this is for sure a breaking change. There may be an automated test pipeline in a customer's infrastructure that is creating bogus API keys (where it doesn't matter) but would suddenly break due to this change, or some other edge case I'm not thinking of.  

@jfreden my (weak) preference would be to go for (2) -- consistency is good, but the Create API key API behavior feels essentially like an oversight/bug so not necessarily worth inheriting. Just my 2 cents though, I don't feel strongly here!

I definitely don't think it's worth going through the breaking changes process.

**Comment by jakelandis** [comment]
I would also have (weak) preference for (2) assuming there are no BWC concerns. I can see testing usecase to create a pre-expired API key, but much harder to rationalize a usecase for pre-expiring while updating an API key.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java [hunk 1]
```diff
@@ -338,13 +339,13 @@ && hasRemoteIndices(request.getRoleDescriptors())) {
...
                 return;
             }
-            final IllegalArgumentException workflowsValidationException = validateWorkflowsRestrictionConstraints(
-                transportVersion,
-                request.getRoleDescriptors(),
-                userRoleDescriptors
-            );
-            if (workflowsValidationException != null) {
-                listener.onFailure(workflowsValidationException);
+            List<IllegalArgumentException> failedValidations = Stream.of(
```
[the remark this answers is not in the document]
Comment (jfreden) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java]: Yes, good point. I've moved it. Thank you!

**Comment by jfreden** [comment]
Thanks for the input @jakelandis and @n1v0lg ! I've changed it to (2).

**Review by n1v0lg — APPROVED** [review]
LGTM! Thanks for iterating on this.

**Pull request merged**